Framework/Core/src/runDataProcessing.cxx

```diff
@@ -2262,6 +2262,11 @@ int doMain(int argc, char** argv, o2::framework::WorkflowSpec const& workflow,
   // check if DataProcessorSpec at i depends on j
   auto checkDependencies = [& workflow = physicalWorkflow](int i, int j) {
     DataProcessorSpec const& a = workflow[i];
+    // The output proxy is always last so that we
+    // can do the forwarding.
+    if (a.name == "dpl-output-proxy") {
```

Comment:
The only question would be if we have another condition to check. If we change the name, which is defined some other place, this might break the logic

Reply:
I was thinking we could introduce some sort of "sink dataprocessor" label to attach to the `DataProcessorSpec`. However that would have the problem that the devices carrying that label in a topology must not share inputs.

That said, I think the output proxy is the only one which needs to behave that way, so I would say that for now the hardcoded check is "sustainable".